library/cli.py

```diff
@@ -1,33 +1,47 @@
 import os
 import time
+from typing import Optional
 
 import typer
 
-app = typer.Typer()
-
-
-@app.command()
-def ingest(name: str, config: str):
-    """
-    ingest a dataset from source to destination
-    """
-    typer.echo(f"ingesting {name} using {config} ...")
-
-
-@app.command()
-def show(name: str):
-    """
-    show versions available for given dataset
-    """
-    typer.echo(f"showing {name} ...")
+from .archive import Archive
 
+app = typer.Typer()
 
+# fmt: off
 @app.command()
-def load(name: str):
+def archive(
+    path: str = typer.Option(None, "--path", "-f", help="Path to config yml"),
+    output_format: str = typer.Option("pgdump", "--output-format", "-o", help="csv, geojson, shapefile, pgdump and postgres"),
+    push: bool = typer.Option(False, "--s3", "-s", help="Push to s3"),
+    clean: bool = typer.Option(False, "--clean", "-c", help="Remove temporary files"),
+    latest: bool = typer.Option(False, "--latest", "-l", help="Tag with latest"),
+    name: str = typer.Option(None, "--name", "-n", help="Name of the dataset, if supplied, \"path\" will ignored"),
+    compress: bool = typer.Option(False, "--compress", help="Compress output"),
+    inplace: bool = typer.Option(False, "--inplace", help="Only keeping zipped file"),
+    postgres_url: str = typer.Option(None, "--postgres-url", help="Postgres connection url"),
+    version: str = typer.Option(None, "--version", "-v", help="Custom version input"),
+) -> None:
+# fmt: on
     """
-    load a dataset from s3/database to a build database
+    Archive a dataset from source to destination
```

Comment:
It might help if we added a bit more detail about where the parameters get passed. "Optional parameters get passed to an Archive object, which finds a config file using either the file path or dataset name. Then, the parameters and config path get passed to the ingestor of the desired output format to read and transform the data for final loading to s3" Maybe something like that.

Reply:
Also just learned that typer supports undefined parameters to be passed in similar to **kwargs.

Reply:
I think I'm going to wait before fulling implement the documentation for this, because they will be subjected to frequent changes